R/system.R

```diff
@@ -1437,20 +1437,45 @@ download_data_file <- function(url, type){
   }
 }
 
-#'Wrapper for readxl::read_excel to support remote file
+#'Wrapper for openxlsx::read.xlsx (in case of .xlsx file) and readxl::read_excel (in case of old .xls file)
+#'Use openxlsx::read.xlsx since it's memory footprint is less than that of readxl::read_excel and this creates benefit for users with less memory like Windows 32 bit users.
 #'@export
-read_excel_file <- function(path, sheet = 1, col_names = TRUE, col_types = NULL, na = "", skip = 0, trim_ws = TRUE, n_max = Inf){
-  loadNamespace("readxl")
-  loadNamespace("stringr")
-  if (stringr::str_detect(path, "^https://") ||
-      stringr::str_detect(path, "^http://") ||
-      stringr::str_detect(path, "^ftp://")) {
-    tmp <- download_data_file(path, "excel")
-    readxl::read_excel(tmp, sheet = sheet, col_names = col_names, col_types = col_types, na = na, trim_ws = trim_ws, skip = skip, n_max = n_max)
-  } else {
-    # if it's local file simply call readxl::read_excel
-    readxl::read_excel(path, sheet = sheet, col_names = col_names, col_types = col_types, na = na, trim_ws = trim_ws, skip = skip, n_max = n_max)
+read_excel_file <- function(path, sheet = 1, col_names = TRUE, col_types = NULL, na = "", skip = 0, trim_ws = TRUE, n_max = Inf, detectDates = FALSE, skipEmptyRows = FALSE, skipEmptyCols = FALSE, check.names = FALSE, ...){
+  loadNamespace("openxlsx")
+  loadNamespace('readr')
+  loadNamespace('stringr')
+  df <- NULL
+  # for .xlsx file extension
+  if(stringr::str_detect(path, '.xlsx')) {
+    if(n_max != Inf) {
+      df <- openxlsx::read.xlsx(xlsxFile = path, rows=skip+1:n_max, sheet = sheet, colNames = col_names, na.strings = na, skipEmptyRows = skipEmptyRows, skipEmptyCols = skipEmptyCols , check.names = check.names, detectDates = detectDates)
```

Comment:
I think you need parenthesis around skip+1 as I doubted in my previous comment.
![image](https://user-images.githubusercontent.com/19518/43220776-bc469692-8fff-11e8-9c6e-e95e6b765c12.png)

Reply:
changed as suggested